File: snippets/rust/src/fiat_currencies.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;

async fn list_supported_fiat_currencies(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: list-fiat-currencies
    let supported_fiat_currencies = sdk.list_fiat_currencies().await?;
    // ANCHOR_END: list-fiat-currencies

    Ok(())
}

async fn get_current_rates(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: fetch-fiat-rates
    let fiat_rates = sdk.fetch_fiat_rates().await?;
    // ANCHOR_END: fetch-fiat-rates

    Ok(())
}

File: snippets/react-native/fiat_currencies.ts
import {
  listFiatCurrencies,
  fetchFiatRates
} from '@breeztech/react-native-breez-sdk'

const exampleListCurrencies = async () => {
  // ANCHOR: list-fiat-currencies
  try {
    const fiatCurrencies = await listFiatCurrencies()
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: list-fiat-currencies
}

const exampleFetchRates = async () => {
  // ANCHOR: fetch-fiat-rates
  try {
    const fiatRates = await fetchFiatRates()
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: fetch-fiat-rates
}
